homeassistant/components/risco/config_flow.py

```diff
@@ -199,6 +217,10 @@ def _options_schema(self):
                     CONF_CODE_DISARM_REQUIRED,
                     default=self._data[CONF_CODE_DISARM_REQUIRED],
                 ): bool,
+                vol.Required(
+                    CONF_COMMUNICATION_DELAY,
+                    default=self._data[CONF_COMMUNICATION_DELAY],
```

Comment:
Can't we start without an option and just have a default for now?

Reply:
I've made some changes according to your remarks. I hope I interpreted your suggestions well.

Reply:
My question and suggestion was to remove the option for now and just store the automatically found setting in the config entry options.

Reply:
I've noticed on my system that sometimes the integration is set up with 0 seconds delay. This means that sometimes the panel replies from the first attempt without any delay. This also means subsequent restarts often fail because most of the time it needs at least a one second delay. To me, changing it via the options seems to be the most straight forward way of fixing it manually to get the delay setup correctly.

Reply:
Can't we try the iterative approach when the connection fails during config entry setup too before raising `ConfigEntryNotReady`?

Reply:
The iterative approach would introduce an additional delay of 5 seconds for my system and it already takes 17 seconds to initialize with the current implementation.
How about doing an iterative approach during setup and store the detected delay + 1 second. This way it will 100% succeed for any system I suppose without the need to use a configurable option. This would also imply that it introduces an additional 1 second delay for newly setup systems. This is something OnFreud wanted to avoid but this in favor of the user friendliness.

If we use this approach, I guess I can store the communication_delay parameter back into the data (without UI)? Or should it stay in the options?

Reply:
The iterative approach only comes into play if we fail to connect right? I don't see a problem with taking some time to find the correct connection settings if we fail to connect the first time.

Why do we need to add 1 second to the detected delay?

Reply:
> The iterative approach only comes into play if we fail to connect, right? I don't see a problem with taking some time to find the correct connection settings if we fail to connect the first time.

Correct, but for my system let's say 8 out of 10 times it will fail on the first attempt causing this additional delay of 6 seconds in my case (5 reset + 1 delay). Maybe there are even slower systems out there in the wild which could need an even longer delay so it can go up to 13 or 21 seconds. This means that worst case the initialization can take up to 21 seconds of attempts + 17 seconds initialization = 38 seconds to just initialize the integration while it is known to need an additional delay.

> Why do we need to add 1 second to the detected delay?

Like mentioned above, 8 out of 10 times it works without any additional delay while it always works with a delay of 1 second. Let's say we don't go for the iterative approach during initialization then we need to make sure we store a value that will always work during initialization. So for my system, it SOMETIMES responds without a delay during setup. If we add an additional second to the delay we detect, we know it will always work.

Reply:
The long time to set up the integration will only be done once and only of the connect attempt fails the first time. After that we know the needed delay, until there's another connect failure.

A 1 second addition to the detected delay as buffer seems reasonable to me.

Reply:
Instead of a 1 second addition to the detected buffer, which would result in a 1 second delay for everyone, we can repeat this process every time the config entry is loaded, starting with the saved value instead of 0. Upon success, if the new value is higher than the saved one, we save it.

Reply:
This also leave the potential to save a value that might be to high, for example after a power outage or a temporary network interruption. Because of that reason I'm not a big fan of saving a new value once the entry has been created.

Reply:
That would be very rare (because it still has to succeed within a short time frame), and the impact would be a slightly longer delay for one user. I think it's better than a guaranteed longer delay for every user.

Reply:
I've updated the connect strategy. If we can agree on this implementation, I'll update and add the tests.